core/src/test/scala/org/apache/spark/AccumulatorSuite.scala

```diff
@@ -209,10 +209,8 @@ class AccumulatorSuite extends SparkFunSuite with Matchers with LocalSparkContex
     System.gc()
     assert(ref.get.isEmpty)
 
-    // Getting a garbage collected accum should throw error
-    intercept[IllegalStateException] {
-      AccumulatorContext.get(accId)
-    }
+    // Getting a garbage collected accum should return None.
+    assert(AccumulatorContext.get(accId).isEmpty)
```

Comment:
Do we have a way to simulate the scenario in description by a new test?

Reply:
Added former exception message as a warning.

Reply:
Unfortunately I haven't been able to reproduce the required sequence of events in tests. Let me collect more info from failed prod jobs.

Reply:
Added a test.